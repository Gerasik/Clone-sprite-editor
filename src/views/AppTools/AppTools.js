export default class AppTools {
  constructor(className, parentBlock) {
    this.className = className;
    this.parentBlock = parentBlock;
    this.block = null;
  }

  render() {
    const tools = document.createElement('section');
    const toolsBar = document.createElement('div');
    const toolList = [{
      name: 'pen',
      icon: 'f304;',
    }, {
      name: 'fill',
      icon: 'f576',
    }, {
      name: 'eraser',
      icon: 'f12d',
    }, {
      name: 'line',
      icon: 'f304',
    }, {
      name: 'square-fill',
      icon: 'f45c',
    }, {
      name: 'square',
      icon: 'f0c8',
    }, {
      name: 'circle-fill',
      icon: 'f111',
    }, {
      name: 'circle',
      icon: 'f111',
    }, {
      name: 'light',
      icon: 'f067',
    }, {
      name: 'dark',
      icon: 'f068',
    }, {
      name: 'color-pick',
      icon: 'f1fb',
    }, {
      name: 'rotate',
      icon: 'f0e2',
    }, {
      name: 'move',
      icon: 'f0b2',
    }, {
      name: 'mirror',
      icon: 'f071',
    }];
    tools.classList.add(this.className);
    toolsBar.classList.add(`${this.className}__bar`);
    toolList.forEach(elem => AppTools.addTool(elem, toolsBar, this.className));
    tools.appendChild(toolsBar);
    this.parentBlock.appendChild(tools);
    return { tools, toolsBar };
  }

  static createInputColor(name, color) {
    const inputColor = document.createElement('input');
    inputColor.setAttribute('type', 'color');
    inputColor.setAttribute('id', name);
    inputColor.setAttribute('name', name);
    inputColor.setAttribute('value', color);
    return inputColor;
  }

  static addTool(elem, parent, blockName) {
    const tool = document.createElement('button');
    tool.setAttribute('title', elem.name);
    tool.classList.add(`${blockName}__item`);
    tool.classList.add(`${blockName}__${elem.name}`);
    tool.innerHTML = `<span>&#x${elem.icon}</span>`;
    parent.appendChild(tool);
  }

  static addcolorList(parent, colors) {
    const colorList = document.createElement('div');
    colorList.classList.add('tools__color-list');
    for (let i = 0; i < colors.length; i += 1) {
      if (!(i === 0)) {
        const swap = document.createElement('span');
        swap.innerHTML = '&#xf362';
        colorList.appendChild(swap);
      }
      colorList.appendChild(AppTools.createInputColor(colors[i].name, colors[i].color));
    }
    parent.appendChild(colorList);
    return colorList;
  }

  static canvasSizeList(parent) {
    const block = document.createElement('div');
    const name = document.createElement('span');
    name.innerHTML = 'Canvas size:';
    const select = document.createElement('select');
    const opt1 = document.createElement('option');
    opt1.innerHTML = '32x32';
    opt1.setAttribute('selected', 'selected');
    const opt2 = document.createElement('option');
    opt2.innerHTML = '64x64';
    const opt3 = document.createElement('option');
    opt3.innerHTML = '128x128';
    select.appendChild(opt1);
    select.appendChild(opt2);
    select.appendChild(opt3);
    block.appendChild(name);
    block.appendChild(select);
    parent.appendChild(block);
    return block;
  }

  static cursorPosition() {
    const block = document.createElement('div');
    const x = document.createElement('div');
    const y = document.createElement('div');
    block.classList.add('cursorPositon');
    x.innerHTML = '<span>X position:</span><span id="x">0</span>';
    y.innerHTML = '<span>Y position:</span><span id="y">0</span>';
    block.appendChild(x);
    block.appendChild(y);
    document.body.appendChild(block);
  }
}
