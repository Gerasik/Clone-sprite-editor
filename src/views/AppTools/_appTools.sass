$main-color: black
$bgc-color: #fff
.tools
  padding: 5px
  margin: 10px
  max-width: 100px
  display: flex
  flex-flow: column
  background-color: $bgc-color
  &__active
    box-shadow: inset 0 0 6px -1px black
  &__item
    padding: 5px
    margin: 5px
    width: 40px
    height: 40px
    cursor: pointer
    span
      font-family: 'Awesome'
      font-size: 24px
      color: $main-color
  &__square
    span
      color: transparent
      border: 3px solid black
  &__circle
    span
      border-radius: 50%
      color: transparent
      border: 3px solid black
  &__mirror
    position: relative
    &:before
      content: ''
      position: absolute
      background-color: #EFEFEF
      width: 5px
      height: 90%
      top: 0
      left: 16px
  &__line
    span
      border-bottom: 4px solid $main-color
  &__bar
    border-bottom: 1px solid $main-color
    margin-bottom: 5px
    padding-bottom: 5px
  &__color-list
    display: flex
    justify-content: space-around
    input[type='color']
      background-color: #fff
      width: 40px
      border: inherit
      padding: 0
      margin: 0
    span
      font-family: 'Awesome'
      font-size: 22px
      color: $main-color
      cursor: pointer
.cursorPositon
  position: fixed
  bottom: 20px
  right: 20px