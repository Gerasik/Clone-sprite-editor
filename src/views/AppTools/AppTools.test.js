import AppTools from './AppTools';

describe('new AppTolls', () => {
  it('Should be an instance of Object', () => {
    expect(new AppTools('tools', document.body)).toBeInstanceOf(Object);
  });
  it('Should be contain right parameters', () => {
    const newObj = new AppTools('tools', document.body);
    expect(newObj.className).toBe('tools');
    expect(newObj.parentBlock).toBe(document.body);
    expect(newObj.block).toBe(null);
  });
});

describe('AppTolls.prototype.render', () => {
  it('Should be an instance of Function', () => {
    expect(AppTools.prototype.render).toBeInstanceOf(Function);
  });

  it('Should be render slide', () => {
    const context = {
      parentBlock: document.body,
    };
    AppTools.prototype.render.call(context);
    expect(document.body.innerHTML).toMatchSnapshot();
  });
});

describe('AppTolls.prototype.addTool', () => {
  it('Should be set right attribute', () => {
    const context = {
      name: 'text',
      color: 'red',
    };
    const ans = AppTools.createInputColor(context.name, context.color);
    expect(ans.getAttribute('id')).toBe(context.name);
    expect(ans.getAttribute('value')).toBe(context.color);
  });
});

describe('AppTolls.prototype.canvasSizeList', () => {
  it('Should be set right attribute', () => {
    const ans = AppTools.canvasSizeList(document.body);
    expect(ans.getElementsByTagName('option')[0].innerHTML).toBe('32x32');
    expect(ans.getElementsByTagName('option')[1].innerHTML).toBe('64x64');
    expect(ans.getElementsByTagName('option')[2].innerHTML).toBe('128x128');
  });
});

describe('AppTolls.prototype.cursorPosition', () => {
  it('Should be an instance of Function', () => {
    expect(AppTools.prototype.render).toBeInstanceOf(Function);
  });

  it('Should be render slide', () => {
    AppTools.cursorPosition();
    expect(document.body.innerHTML).toMatchSnapshot();
  });
});

describe('AppTolls.prototype.addcolorList', () => {
  it('Should be set 3 child', () => {
    const color = [{ name: 'color1', color: '#000000' }, { name: 'color2', color: '#fefefe' }];
    const ans = AppTools.addcolorList(document.body, color);
    expect(ans.children.length).toBe(3);
  });
  it('Should be set 5 child', () => {
    const color = [{ name: 'color1', color: '#000000' }, { name: 'color2', color: '#fefefe' }, { name: 'color3', color: '#fddddd' }];
    const ans = AppTools.addcolorList(document.body, color);
    expect(ans.children.length).toBe(5);
  });
});
