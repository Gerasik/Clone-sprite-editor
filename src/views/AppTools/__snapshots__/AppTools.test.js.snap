// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTolls.prototype.cursorPosition Should be render slide 1`] = `"<section class=\\"undefined\\"><div class=\\"undefined__bar\\"><button title=\\"pen\\" class=\\"undefined__item undefined__pen\\"><span></span></button><button title=\\"fill\\" class=\\"undefined__item undefined__fill\\"><span></span></button><button title=\\"eraser\\" class=\\"undefined__item undefined__eraser\\"><span></span></button><button title=\\"line\\" class=\\"undefined__item undefined__line\\"><span></span></button><button title=\\"square-fill\\" class=\\"undefined__item undefined__square-fill\\"><span></span></button><button title=\\"square\\" class=\\"undefined__item undefined__square\\"><span></span></button><button title=\\"circle-fill\\" class=\\"undefined__item undefined__circle-fill\\"><span></span></button><button title=\\"circle\\" class=\\"undefined__item undefined__circle\\"><span></span></button><button title=\\"light\\" class=\\"undefined__item undefined__light\\"><span></span></button><button title=\\"dark\\" class=\\"undefined__item undefined__dark\\"><span></span></button><button title=\\"color-pick\\" class=\\"undefined__item undefined__color-pick\\"><span></span></button><button title=\\"rotate\\" class=\\"undefined__item undefined__rotate\\"><span></span></button><button title=\\"move\\" class=\\"undefined__item undefined__move\\"><span></span></button><button title=\\"mirror\\" class=\\"undefined__item undefined__mirror\\"><span></span></button></div></section><div><span>Canvas size:</span><select><option selected=\\"selected\\">32x32</option><option>64x64</option><option>128x128</option></select></div><div class=\\"cursorPositon\\"><div><span>X position:</span><span id=\\"x\\">0</span></div><div><span>Y position:</span><span id=\\"y\\">0</span></div></div>"`;

exports[`AppTolls.prototype.render Should be render slide 1`] = `"<section class=\\"undefined\\"><div class=\\"undefined__bar\\"><button title=\\"pen\\" class=\\"undefined__item undefined__pen\\"><span></span></button><button title=\\"fill\\" class=\\"undefined__item undefined__fill\\"><span></span></button><button title=\\"eraser\\" class=\\"undefined__item undefined__eraser\\"><span></span></button><button title=\\"line\\" class=\\"undefined__item undefined__line\\"><span></span></button><button title=\\"square-fill\\" class=\\"undefined__item undefined__square-fill\\"><span></span></button><button title=\\"square\\" class=\\"undefined__item undefined__square\\"><span></span></button><button title=\\"circle-fill\\" class=\\"undefined__item undefined__circle-fill\\"><span></span></button><button title=\\"circle\\" class=\\"undefined__item undefined__circle\\"><span></span></button><button title=\\"light\\" class=\\"undefined__item undefined__light\\"><span></span></button><button title=\\"dark\\" class=\\"undefined__item undefined__dark\\"><span></span></button><button title=\\"color-pick\\" class=\\"undefined__item undefined__color-pick\\"><span></span></button><button title=\\"rotate\\" class=\\"undefined__item undefined__rotate\\"><span></span></button><button title=\\"move\\" class=\\"undefined__item undefined__move\\"><span></span></button><button title=\\"mirror\\" class=\\"undefined__item undefined__mirror\\"><span></span></button></div></section>"`;
